lib/util/binarySearchBounds.js

```diff
@@ -8,6 +8,28 @@
 /* cspell:disable-next-line */
 // Refactor: Peter Somogyvari @petermetz
 
+/** @typedef {">=" | "<=" | "<" | ">" | "-" } BinarySearchPredicate */
+/** @typedef {"GE" | "GT" | "LT" | "LE" | "EQ" } SearchPredicateSuffix */
+
+/**
+ * Helper function for compiling binary search functions.
+ *
+ * The generated code uses a while loop to repeatedly divide the search interval
+ * in half until the desired element is found, or the search interval is empty.
+ *
+ * The following is an example of a generated function for calling `compileSearch("myFunc", ">=", true, ["value"], false)`:
+ *
+ * ```js
+ * function myFunc(a,l,h,value){var i=l-1;while(l<=h){var m=(l+h)>>>1,x=a[m];if(>=){i=m;l=m+1}else{h=m-1}}return i};
```

Comment:
This sample is malformed. See the `if(>=)` bit.

Reply:
I ran this directly via Node. 😫 I'll double check the generation.

Reply:
Corrected.